sql/src/main/java/io/crate/analyze/WhereClause.java

```diff
@@ -201,4 +202,23 @@ public int hashCode() {
     public boolean hasVersions() {
         return docKeys.isPresent() && docKeys.get().withVersions();
     }
+
+
+    /**
+     * Adds another query to this WhereClause.
+     *
+     * The result is either a new WhereClause or the same (but modified) instance.
+     */
+    public WhereClause add(Symbol otherQuery) {
+        assert !docKeys.isPresent() : "Cannot add otherQuery if there are docKeys in the WhereClause";
+        if (this == MATCH_ALL) {
```

Comment:
dunno if checking against the singleton is always enough, probably use the query field?